src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,89 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { addComment } from '../services/comments';
+
+type Props = {
+  selectedPost: Post | null;
+  addNewComment: (newComment: Comment) => void;
+  handleSetError: () => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  addNewComment,
+  handleSetError,
+}) => {
+  const [queryName, setQueryName] = useState('');
+  const [queryEmail, setQueryEmail] = useState('');
+  const [queryComment, setQueryComment] = useState('');
```

Comment:
make one state and one handler for all form fields. use event.target.name for change proper form field.
```suggestion
  const [newComment, setNewComment] = useState({
    name: '',
    email: '',
    comment: '',
  });
```
attibute name should be the same as each newComment field ('name', 'email', 'comment')